src/bip44.c

```diff
@@ -247,46 +313,94 @@ size_t bip44_printToStr(const bip44_path_t* pathSpec, char* out, size_t outSize)
 	return ptr - out;
 }
 
-
-bip44_path_type_t bip44_classifyPath(const bip44_path_t* pathSpec)
+static bip44_path_type_t bip44_classifyOrdinaryWalletPath(const bip44_path_t* pathSpec)
 {
-	if (bip44_hasValidCardanoPoolColdKeyPrefix(pathSpec)) {
-		if (bip44_isValidPoolColdKeyPath(pathSpec)) {
-			return PATH_POOL_COLD_KEY;
-		} else {
+	ASSERT(bip44_hasOrdinaryWalletKeyPrefix(pathSpec));
+
+	switch (pathSpec->length) {
+	case 3: {
+		return isHardened(bip44_getAccount(pathSpec)) ?
+		       PATH_ORDINARY_ACCOUNT :
+		       PATH_INVALID;
+	}
+	case 5: {
+		const uint8_t chainType = bip44_getChainTypeValue(pathSpec);
+		switch (chainType) {
+
+		case CARDANO_CHAIN_INTERNAL:
+		case CARDANO_CHAIN_EXTERNAL:
+			// we do not exclude hardened address index
```

Comment:
I see. But for some reason the comment is still confusing. It's not obvious to me why we should exclude it. Maybe the comment could say `We don't need to check whether the address index is hardened or not as both are allowed. However, a warning is shown for addresses with hardened address index in certain places as it is unusual and should not be used.` Or something like that. It should explain why we don't exclude them. However if you think it's OK, I'm good with it too 😉

Reply:
I have replaced the comment with 
			// We do not exclude hardened address index for legacy reasons;
			// such indices have been allowed since Byron and no one really knows if they are in use,
			// so we don't want to make users' funds on such addresses unavailable.
			// But such addresses are given a warning
			// and are never hidden from users (see bip44_isPathReasonable).
which is much more clear.

Also, hardened address index need not to be allowed for multisig since the "legacy reasons" argument does not apply, we never supported such paths.